test/test_python_dispatch.py

```diff
@@ -0,0 +1,139 @@
+import torch
+from torch.testing._internal.common_utils import TestCase, run_tests
+from torch._C import _disabled_torch_function_impl
+
+from typing import Iterator, List, Dict
+import logging
+import contextlib
+
+# TODO: TensorBase should work
+class LoggingTensor(torch.Tensor):
+    elem: torch.Tensor
+
+    @staticmethod
+    def __new__(cls, elem, *args, **kwargs):
+        r = torch.Tensor._make_subclass(cls, elem.to('meta'), elem.requires_grad)
+        r.elem = elem
+        return r
+
+    def __repr__(self):
+        return f"LoggingTensor({self.elem})"
+
+    def __str__(self):
+        return f"LoggingTensor({self.elem})"
+
+    def __format__(self, format_spec):
+        return f"LoggingTensor({self.elem})"
+
+    __torch_function__ = _disabled_torch_function_impl
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
```

Comment:
Can you go through an example of what exactly the chain of dispatch is when one calls e.g. torch.sin on a LoggingTensor? Something like an updated version of #59049's "How does holder-less dispatch to Python work" section would be helpful (do we still dispatch to `__torch_function__` first, then that enters the dispatcher, and then that comes around to `__torch_dispatch__` ?

Reply:
I'll put this in code somewhere, but just quickly:

1. Call `torch.sin`
2. Attempt `__torch_function__`. In LoggingTensor torch function is disabled so we bypass it entirely
3. Enter dispatcher, wind your way through Autograd
4. Hit Python dispatch key, call `__torch_dispatch__`